app/javascript/packs/react-bootloader.jsx

```diff
@@ -10,8 +10,15 @@ export const initReact = (mappings) => {
 }
 
 const render = (elem, component) => {
-  ReactDOM.render(<React.StrictMode>{component}</React.StrictMode>, elem)
-  elem.classList.add('--hydrated')
+  ReactDOM.render(
+    <React.StrictMode>{component}</React.StrictMode>,
+    elem,
+    () => {
+      setTimeout(() => {
+        elem.classList.add('--hydrated')
```

Comment:
If you really want this and not be bound to a flaky `setTimeout`, move it into the component.

```ts
function MyComponent() {
  useLayoutEffect(() => {
  	ref.current!.parentElement.classList.add('--hydrated')

	// technically you could even un-hydrate it on unmount:
	const parent = ref.current!.parentElement
    return (): void => {
      parent.classList.remove('--hydrated')
    }
  }, [])
}

Reply:
Ah, ok. That's really helpful to know. Will save that code for other places. Thanks.